### PR TITLE
Using ament_target_dependencies where possible

### PR DESCRIPTION
**Pull request by ivanpauno**
Partially targets ros2/ros2#658

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -35,9 +35,13 @@ set(${PROJECT_NAME}_SRCS
   src/multi_library_class_loader.cpp
 )
 add_library(${PROJECT_NAME} SHARED ${${PROJECT_NAME}_SRCS})
-target_include_directories(${PROJECT_NAME}
-  PUBLIC include ${console_bridge_INCLUDE_DIRS} ${Poco_INCLUDE_DIRS})
-target_link_libraries(${PROJECT_NAME} ${console_bridge_LIBRARIES} ${Poco_LIBRARIES})
+if(ament_cmake_FOUND)
```
Comment (nuclearsandwich) [inline CMakeLists.txt:38]: Is this conditional correct? This reads to me like `ament_target_dependencies` would be used if ament_cmake is not found.

File: CMakeLists.txt [hunk 2]
```diff
@@ -35,9 +35,13 @@ set(${PROJECT_NAME}_SRCS
...
+  target_include_directories(${PROJECT_NAME}
+    PUBLIC include ${console_bridge_INCLUDE_DIRS} ${Poco_INCLUDE_DIRS})
+  target_link_libraries(${PROJECT_NAME} ${console_bridge_LIBRARIES} ${Poco_LIBRARIES})
+else()
+  ament_target_dependencies(${PROJECT_NAME} "console_bridge" "Poco")
```
Comment (nuclearsandwich) [inline CMakeLists.txt]: Does ament_target_dependencies support non-ament packages as dependencies?

Comment (dirk-thomas) [reply CMakeLists.txt]: The API docs of the CMake function should answer the question: https://github.com/ament/ament_cmake/blob/207a07b72af1e5a780887d7c2268df57ef456af3/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake#L19-L22

File: CMakeLists.txt (hunk 1 above)
Comment (ivanpauno) [reply CMakeLists.txt:38]: Wrong if/else order. Corrected

**Review by nuclearsandwich — APPROVED** [review]
:+1: Please link the CI builds (a link to the tracking issue comment is fine) when you trigger them.

**Comment by ivanpauno** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/6450/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2821/)
* macOS [Build Status](https://ci.ros2.org/job/ci_osx/5296/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6270/)

All failures are unrelated. 
@nuclearsandwich This can be merged (I don't have enough permissions here).

**Pull request merged**